app/models/carrier.rb

```diff
@@ -0,0 +1,13 @@
+class Carrier < ActiveRecord::Base
+  validates :request, presence: true
+  validates :response, presence: true
+
+  def self.create_by(request, response)
+    carrier = Carrier.new
+    carrier.request  = request.body.read
+    carrier.response = response.body
+    if carrier.save
+      return carrier 
+    end 
+  end 
```

Comment:
Interesting to me that you saved the whole request and response as JSON strings. Cool way to think about this.